src/hotspot/share/runtime/mutexLocker.hpp

```diff
@@ -28,6 +28,7 @@
 #include "memory/allocation.hpp"
 #include "runtime/flags/flagSetting.hpp"
 #include "runtime/mutex.hpp"
+#include "runtime/thread.hpp"
```

Comment:
This include is not needed because there are no uses that require the definition of Thread.

After pull, I am getting circularity errors from runtime/thread.hpp -> jfrThreadLocal.hpp (which now needs runtime/mutexLocker.hpp -> runtime/thread.hpp

Reply:
>This include is not needed because there are no uses that require the definition of Thread.

Right, seems like the forward declaration used to be provided by `memory/allocation.hpp`. Let's get rid of the include and use a forward declaration of our own instead.

Reply:
Should I open another PR to remove it? I can also remove the locking related to NMT in arena.cpp in the same PR as well (as per [this comment](https://github.com/openjdk/jdk/pull/20852#issuecomment-2450515494)).

Reply:
I've fixed that particular bug and integrated it already :-).